helix-term/src/commands.rs

```diff
@@ -584,6 +585,34 @@ fn goto_first_nonwhitespace(cx: &mut Context) {
     doc.set_selection(view.id, selection);
 }
 
+fn trim_selections(cx: &mut Context) {
+    let (view, doc) = current!(cx.editor);
+    let text = doc.text().slice(..);
+
+    let selection = doc.selection(view.id).clone().transform(|range| {
+        let mut start = range.from();
+        let mut end = range.to();
+
+        start = movement::skip_while(text, start, |x| x.is_whitespace()).unwrap_or(start);
+        if end > 0 {
```

Comment:
Rather than this check for > 0 then -1 I think you have to useprev_grapheme_boundary

Reply:
If backwards_skip_while returns none then you also end up increasing end by 1 here (end = end+1)

Reply:
Actually, I think you want to use range.cursor https://github.com/helix-editor/helix/blob/b824e091a948c076a428fb981cd5be2929378533/helix-core/src/selection.rs#L264

This way you have an inclusive range between range.anchor and range.cursor()(you have to check which one is start and which one is end based on orientation though)